File: Problem2/code.js
let fib = (n, memo = []) =>{
    if(memo[n] != null){
        return memo[n]
    }

    let result
    if( n <= 2){
        result = 1
    } else {
        result = fib(n - 1, memo) + fib(n - 2, memo)
    }
    memo[n] = result

    return result
}

let x = 0

let evenFibs = []

while(fib(x) < 4000000){
    if(fib(x) % 2 === 0){
        evenFibs.push(fib(x))
    }
    x++
}

let evenFibsSum = evenFibs.reduce((a, b) => a + b, 0)

console.log(evenFibsSum)

File: Problem2/code.py
import functools

@functools.cache
def fib(n):
    if n <= 2:
        return 1
    else: 
        return fib(n - 1) + fib(n-2)
        
a = 2

evenFibs = []

while(fib(a) < 4000000):
    if fib(a) % 2 == 0:
        evenFibs.append(fib(a))
    a+=1

print(sum(evenFibs))
